site/src/pages/WorkspacePage/WorkspaceActions/DownloadLogsDialog.tsx

```diff
@@ -126,20 +188,70 @@ export const DownloadLogsDialog: FC<DownloadLogsDialogProps> = ({
   );
 };
 
+type DownloadingItemProps = Readonly<{
+  // A value of undefined indicates that the component will wait forever
+  giveUpTimeMs?: number;
+  file: DownloadableFile;
+}>;
+
+const DownloadingItem: FC<DownloadingItemProps> = ({ file, giveUpTimeMs }) => {
+  const theme = useTheme();
+  const [isWaiting, setIsWaiting] = useState(true);
+  useEffect(() => {
+    if (giveUpTimeMs === undefined || file.blob !== undefined) {
+      setIsWaiting(true);
+      return;
+    }
+
+    const timeoutId = window.setTimeout(
+      () => setIsWaiting(false),
+      giveUpTimeMs,
+    );
+
+    return () => window.clearTimeout(timeoutId);
+  }, [giveUpTimeMs, file]);
+
+  return (
+    <li css={styles.listItem}>
+      <span
+        css={[
+          styles.listItemPrimary,
+          !isWaiting && { color: theme.palette.text.disabled },
+        ]}
+      >
+        {file.name}
+      </span>
+
+      <span css={styles.listItemSecondary}>
+        {file.blob ? (
+          humanBlobSize(file.blob.size)
+        ) : isWaiting ? (
+          <Skeleton variant="text" width={48} height={12} />
+        ) : (
+          <div css={styles.notAvailableText}>
+            <span aria-hidden>
+              <ErrorIcon fontSize="inherit" />
+            </span>
+
+            <p>N/A</p>
+          </div>
+        )}
+      </span>
+    </li>
+  );
+};
+
 function humanBlobSize(size: number) {
-  const units = ["B", "KB", "MB", "GB", "TB"];
   let i = 0;
-  while (size > 1024 && i < units.length) {
+  while (size > 1024 && i < BLOB_SIZE_UNITS.length) {
```

Comment:
Since the `BLOB_SIZE_UNITS`  is only used here, I would not move it to leave outside of the function where it is used.